js/sdk/src/components/EmailCapture.js

```diff
@@ -57,51 +52,91 @@ const EmailCaptureSchema = Yup.object().shape(
+  const { enqueueSnackbar } = useSnackbar()
+  const [usingMagicWallet, setUsingMagicWallet] = useState(false)
+  const [user, setUser] = useState(undefined)
   const { publicKey, connected } = wallet
-  const { submitEmailRequest } = useContext(Nina.Context)
+  const { submitEmailRequest, getVerificationsForUser, verificationState } =
+    useContext(Nina.Context)
   const [open, setOpen] = useState(false)
   const [showSuccessInfo, setShowSuccessInfo] = useState(false)
-  const handleOpen = () => setOpen(true)
-  const handleClose = () => setOpen(false)
   const [formValues, setFormValues] = useState({})
   const [formIsValid, setFormIsValid] = useState(false)
   const [submitButtonText, setSubmitButtonText] = useState('Submit')
-
-  useEffect(() => {
-    if (open) {
-      logEvent('email_request_initiated', 'engagement')
-    }
-  }, [open])
+  const [pending, setPending] = useState(false)
+  const [userVerifications, setUserVerifications] = useState(undefined)
+  const [soundcloudAccount, setSoundcloudAccount] = useState(undefined)
+  const [twitterAccount, settwitterAccount] = useState(undefined)
 
   useEffect(() => {
     if (connected) {
       setFormValues({ ...formValues, wallet: publicKey?.toString() })
+
+      getVerificationsForUser(publicKey?.toString())
     }
   }, [connected, publicKey])
 
+  useEffect(() => {
+    if (verificationState[publicKey?.toString()]) {
+      setUserVerifications(verificationState[publicKey?.toString()])
+    }
+  }, [verificationState, publicKey])
+
+  useEffect(() => {
+    if (wallet.wallet.adapter.name === 'Nina') {
+      setUsingMagicWallet(true)
+      setUser(wallet.wallet.adapter.user)
+    }
+  }, [wallet])
+
   useEffect(() => {
     if (formIsValid) {
       setSubmitButtonText('Submit')
     }
   }, [formValues])
 
+  useEffect(() => {
+    if (userVerifications) {
+      setSoundcloudAccount(
+        getVerificationValue(userVerifications, 'soundcloud')
+      )
+      settwitterAccount(getVerificationValue(userVerifications, 'twitter'))
+    }
+  }, [userVerifications])
+
+  const handleOpen = () => {
+    setOpen(true)
+    logEvent('email_request_initiated', 'engagement')
+    setChildFormOpen(true)
+  }
+
+  const handleClose = () => {
+    setOpen(false)
+    setChildFormOpen(false)
+    setParentOpen(false)
+  }
+
   const handleSubmit = async () => {
     if (formIsValid) {
-      setSubmitButtonText('Submitting...')
+      setPending(true)
       try {
-        await submitEmailRequest(formValues)
+        const request = await submitEmailRequest(formValues)
+        if (request) {
+          setShowSuccessInfo(true)
+          enqueueSnackbar('Email request submitted', { variant: 'success' })
+        }
         logEvent('email_request_success', 'engagement', {
           email: formValues.email,
         })
-        setShowSuccessInfo(true)
       } catch (error) {
         console.warn('email form error', error)
         logEvent('email_request_success', 'engagement', {
```

Comment:
should this be email_request_error or something to that effect since this occurring in the catch block?